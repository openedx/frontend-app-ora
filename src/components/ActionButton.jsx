import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import { Button, StatefulButton } from '@edx/paragon';

import { MutationStatus } from 'constants/index';

export const disabledStates = [MutationStatus.loading];

export const baseClassName = 'w-100 mt-2';
const ActionButton = (props) => {
  const className = classNames(baseClassName, props.className);
  if (!props.onClick && !props.href) {
    return null;
  }
  return props.state
    ? (
      <StatefulButton
        {...props}
        className={className}
        disabledStates={disabledStates}
      />
    ) : (
      <Button {...props} className={className} />
    );
};
ActionButton.defaultProps = {
  state: undefined,
  className: undefined,
  onClick: undefined,
  href: undefined,
};

ActionButton.propTypes = {
  onClick: PropTypes.func,
  state: PropTypes.string,
  className: PropTypes.string,
  href: PropTypes.string,
};

export default ActionButton;
