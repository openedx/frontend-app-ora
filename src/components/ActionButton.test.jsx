import { shallow } from '@edx/react-unit-test-utils';

import ActionButton from './ActionButton';

describe('<ActionButton />', () => {
  const props = {
    state: 'arbitraryState',
  };

  it('render empty when no onClick or href', () => {
    const wrapper = shallow(<ActionButton {...props} />);
    expect(wrapper.isEmptyRender()).toBe(true);
  });

  it('render StatefulButton when state is provided', () => {
    const wrapper = shallow(<ActionButton href="some-href" state="loading" />);
    expect(wrapper.snapshot).toMatchSnapshot();
    expect(wrapper.instance.findByType('StatefulButton')).toHaveLength(1);
    expect(wrapper.instance.findByType('Button')).toHaveLength(0);
  });

  it('render Button when state is not provided', () => {
    const wrapper = shallow(<ActionButton onClick={jest.fn().mockName('onClick')} />);
    expect(wrapper.snapshot).toMatchSnapshot();
    expect(wrapper.instance.findByType('StatefulButton')).toHaveLength(0);
    expect(wrapper.instance.findByType('Button')).toHaveLength(1);
  });
});
