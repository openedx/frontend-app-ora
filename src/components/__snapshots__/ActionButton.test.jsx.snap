// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActionButton /> render Button when state is not provided 1`] = `
<Button
  className="w-100 mt-2"
  onClick={[MockFunction onClick]}
/>
`;

exports[`<ActionButton /> render StatefulButton when state is provided 1`] = `
<StatefulButton
  className="w-100 mt-2"
  disabledStates={
    Array [
      "loading",
    ]
  }
  href="some-href"
  state="loading"
/>
`;
